### ISSUE TITLE
Search favorites

### ISSUE THREAD

**Issue by metmarkosaric** [SUPERCILEX/gnome-clipboard-history#71]
Hi and thanks for this great extension! 

### Problem description

When I search my clipboard history, none of my saved favorites are found. Only the non-favorite entries are found using search.

### Potential solutions/workarounds

Would be nice for the search functionality to also search the favorites.

### Additional context

I like to use the "Only save favorites on disk" option to keep my clipboard clean and it would be great to have an easy way to search through all my favorites and quickly discover the one I'm looking for.

Thanks!

**Comment by SUPERCILEX** [comment]
This issue is similar in nature to #65. Do you have tons of favorites? I intentionally excluded favorites since they're always visible and therefore shouldn't be hard to find.

**Comment by metmarkosaric** [comment]
Thanks for the quick response! Currently I keep no favorites as they're not searchable but ideally I would save 50+ that I use more frequently and everything else would get deleted automatically when the "Only save favorites on disk" is enabled. Anyhow, I'm also happy to use the extension without too many favorites now I'm aware that they are intentionally excluded from search.

**Comment by SUPERCILEX** [comment]
No, that actually sounds reasonable. It shouldn't be hard code wise so why not.

**Comment by metmarkosaric** [comment]
Nice to hear that, thanks!

**Comment by 097115** [comment]
@SUPERCILEX If re-thinking the favourites part, then what would you say about adding an option to may be have them on some sort of separate page? 

Like @metmarkosaric, I would like to have some 50+ favourite entries at my disposal, access them via search, but it's obvious that as things stand now, having that much favourites will simply bury the general clipboard history, making it almost impossible to access the few most recent items with arrows (since you'll have to navigate with the Down key through all the favourites first) :)

**Comment by 097115** [comment]
...Or, on a second thought, may be, not to move favourites into a separate page/list/window, but add an option to keep then in the general list, just never delete them, no matter how old they'd get. Then, they'd be accessible by search just like a regular item, and also won't popup before the history (which could be a bit of inconvenience when there's a lot a favourited items).

What do you think, guys?

**Comment by 097115** [comment]
@SUPERCILEX But that would be another additional hotkey, wouldn't it? :)

While optionally keeping favourites among the other items (for those few users that do have A LOT of saved items) won't slow down accessing either them, or the usual clipboard history in any way?

**Comment by SUPERCILEX** [comment]
Nah, it'd just be a built-in shortcut you can't change since it only works while the UI is open.

Not quite sure what you mean. If you're asking about the performance of having many favorites, I think gnome will start breaking down around several hundred list items.

**Comment by 097115** [comment]
No, I mean that in order to access the recent clipboard history we would (please, correct if I get all wrong):

  1. Press the global hotkey;
  2. Press the additional hotkey you have suggested to skip the favourites;
  3. Proceed with selecting the clipboard history. 

Thus, this additional item (2) arises, if I understand you correctly.

But what if just keep everything pretty much as it is now (and certainly adding this additional hotkey to skip the favourites part wouldn't hurt) but also add an option not to put the saved items separately but to keep them among the usual clipboard history?

  - Now, if we want to quickly exchange some last clipboard history items, we just press the global hotkey and operate with arrows;
  - If we want to search  through the whole clipboard history, nothing changes for us, we press the global hotkey and just search as it is now;
  - And if we want to access our favourites, then again, we just press the global hotkey and search (because with favourites we usually do know what to search, don't we? :) (Of course, given that searching through favourites will be possible :)

Then, even for folks with some dozens of favourites, they still will be easily accessible -- as well as the recent clipboard history, with no intermediate steps (and again, it will be just a niche option, not the new game changing defaults).

**Comment by SUPERCILEX** [comment]
Ohhhh, that's a very neat idea! I'll have to think about it a bit more to decide whether or not the complexity of having two different ways if displaying favorites is worth it. But yeah, cool idea and thanks for the detailed explanation.
